### ISSUE TITLE
Cannot run production build for ejected projects 

### ISSUE THREAD

**Issue by dagi12** [angular/angular-cli#5229]
### Problem
I think title is quite self-explanatory here. Before app ejection I run 
`ng build -prod`
For building production app. Unlike usual build this minify my project and does some other things.
After ejection there is a webpack configuration which as I think corresponds to usual build. There's no webpack configuration for production environment.
### Possible fix
Project ejection should create production webpack configuration that will reflects `ng build -prod` behavior.  And add `webpack --config -p webpack.config.prod.js` to scripts in package.json

**Comment by hansl** [comment]
You can do this by using `ng eject --prod`. The eject command runs with the same flags as the build command.

**Comment by dagi12** [comment]
I've already figure this out. That's why I closed the issue ;)

**Comment by creat-or** [comment]
OK I think don't understand how `eject` meant to be... Why can't I eject the project and have a development AND production environment afterwards?
I thought `eject` is like you do it once and then you cannot use angular-cli features anymore, but you're more flexible in configuration? So how am I supposed to make custom changes (https://github.com/angular/angular-cli/issues/5362#issuecomment-310298275) and build for development and production afterwards?
Thank's for every help!

**Comment by bennett000** [comment]
@creat-or you can it's just how you do so can get opinionated pretty quickly.  

One common approach is to to have an environment variable , like `NODE_ENV` and set it prior to builds.  Then in your webpack config test against it like `const isProd = process.env.NODE_ENV === 'production' ? true : false;`

Then later in your config you can use `isProd` to make choices about what plugins to use etc...

Taking a deeper dive would go waaay off topic.  I bet there are blog posts about this somewhere.

**Comment by khalid-elabbadi** [comment]
i think in an angular-cli ejected project the command `yarn run build` (`npm run build`) must be smart and read arguments `--prod -aot` and pass them to webpack. That way things will run better and we stay on the same path as `ng` build options

**Comment by lifenautjoe** [comment]
Can we re-open this? @dagi12. 

Also whenever you do an `ng-eject --prod` you must manually remove all `package.json` scripts and the `webpack.config.js`file in order for it to work.

**Comment by dagi12** [comment]
Reopened

**Comment by PeterShershov** [comment]
Hey guys, any progress regarding this?

**Comment by vincentjames501** [comment]
+1. Seems like `ng eject` should allow passing similar arguments to `webpack` for generating a production build.

**Comment by hikumealan** [comment]
+1

**Comment by bennett000** [comment]
@hikumealan I think the code coverage in CLI works out of the box, at least with HTML/lcov.  Plus IIRC the karma config is customizable with or _without_ ejecting.

**Comment by GJordan904** [comment]
I ran into this issue after running `ng eject ` and found this issue. It isn't very difficult to work around, just more hassle than it should be and shouldn't require a workaround at all.

IMHO running `ng eject` should spit out a webpack.dev.config and a webpack.prod.config. It should also create npm scripts for build:prod, and watch, in addition to the other scripts it creates.

So until an official solution presents itself, here's my workaround.

1. run `ng eject`
2. rename webpack.config.js to webpack.dev.config.js
3. delete all the npm scripts in package.json
4. run `ng eject --prod`
5. (optional) rename the new webpack.config.js to webpack.prod.config.js
6. setup your npm scripts in package.json
7. run `npm i`

And here is what my npm scripts look like when I'm done.
```
"scripts": {
    "build": "webpack --config webpack.dev.config",
    "watch": "webpack --watch --config webpack.dev.config",
    "build:prod": "webpack --config webpack.prod.config",
    "start": "webpack-dev-server --port=4200 --config webpack.dev.config",
    "test": "karma start ./karma.conf.js",
    "pree2e": "webdriver-manager update --standalone false --gecko false --quiet",
    "e2e": "protractor ./protractor.conf.js"
  }
```

**Comment by phil123456** [comment]
I really dont want to maintain different web pack settings and I need to have custom options in it

why is the eject not creating something that can manage different environements, at least dev/prod

**Comment by clydin** [comment]
eject functionality is no longer available in the latest version (6.0+).  For custom webpack configurations the following unofficial add-on is a suggested solution: https://github.com/manfredsteyer/ngx-build-plus
